pandas/core/internals.py

```diff
@@ -634,11 +637,13 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
 
         if newb.is_numeric and self.is_numeric:
             if newb.shape != self.shape:
-                raise TypeError("cannot set astype for copy = [%s] for dtype "
-                                "(%s [%s]) with smaller itemsize that current "
-                                "(%s [%s])" % (copy, self.dtype.name,
-                                               self.itemsize, newb.dtype.name,
-                                               newb.itemsize))
+                raise TypeError(
+                    "cannot set astype for copy = [{copy}] for dtype "
+                    "({dtype} [{itemsize}]) with smaller itemsize that "
```

Comment:
Is this supposed to be 'smaller itemsize **than** current'? If yes, would you like me to change in this PR?

Reply:
yes